### ISSUE TITLE
HEIF EXIF support

### ISSUE THREAD

**Issue by leslie-wang** [dsoprea/go-exif#8]
Based on apple's reply, https://forums.developer.apple.com/thread/89759, seems like HEIF stores EXIF into mp4 box/atom container. Just wonder if you have any plan to support it?

**Comment by dsoprea** [comment]
Specific image formats are out of scope for this project. It requires you to provide the raw EXIF bytes to process. 

That said, there is also a brute-force method provided that will search for EXIF in a blob of bytes and return the start of the EXIF to the end of the file, which can then be passed to the parser. It is provided for exploratory reasons: https://github.com/dsoprea/go-exif/blob/master/exif.go#L65 . This is what the exif-read-tool command uses to print the EXIF for any file you give it, though it's not gonna handle any encoding/compression if anyone ever introduces that. This tool could be useful in parsing EXIF in any new formats that no tools are yet easily available to parse.

In the interest of providing a complete offering, I also wrote JPEG and PNG parsers to efficiently extract the EXIF information from those formats. 

https://github.com/dsoprea/go-jpeg-image-structure
https://github.com/dsoprea/go-png-image-structure

I did this specifically for the PNG support since it's very new and read/write is not supported by any other libraries/tools.

As a fun exercise, I'd be curious how complicated the HEIF implementation is. It looks like this guy wrote an implementation in Go, and, of course, specifically pushed the EXIF integration to later: https://github.com/strukturag/libheif/blob/master/go/heif/heif.go#L576
